File: commons-security/src/main/java/net/tassia/security/digest/NoParameters.java
package net.tassia.security.digest;

/**
 * An empty parameter class.
 *
 * @since Commons 1.0
 * @author Tassilo
 */
public final class NoParameters extends DigestParameters {

	/**
	 * The singleton instance.
	 */
	public static final NoParameters INSTANCE = new NoParameters();

	/**
	 * Singleton class.
	 */
	private NoParameters() {
	}

}

File: kommons-security/src/commonMain/kotlin/net.tassia.security/digest/impl/NoParameters.kt
package net.tassia.security.digest.impl

import net.tassia.security.digest.DigestParameters

/**
 * The [DigestParameters] to use, when a digest does not actually support any parameters.
 * This is essentially just an empty class.
 *
 * The serialized from of this is an empty byte array.
 *
 * @see NPDigestType
 *
 * @since Kommons 1.0
 * @author Tassilo
 */
object NoParameters : DigestParameters()
